pkg/generate/config/resource.go

```diff
@@ -87,6 +90,26 @@ type ResourceConfig struct {
 	IsARNPrimaryKey bool `json:"is_arn_primary_key"`
 }
 
+// SyncedConfig instructs the code generator on how to generate functions that checks
+// whether a resource was synced or not.
+type SyncedConfig struct {
+	// When is a list of conditions that should be satisfied in order to tell whether a
+	// a resource was synced or not.
+	When []SyncedCondition `json:"when"`
+}
+
+// SyncedCondition represent one of the unique condition that should be fullfiled in
+// order to assert whether a resource is synced.
+type SyncedCondition struct {
+	// Path of the field. e.g Status.Processing
+	Path *string `json:"path"`
+	// Is contains the exact value `Path` should be equal to.
+	Is string `json:"is"`
```

Comment:
I know I'd originally suggested both an `is` and an `in` field, but thinking about this more, we only really need an `in` field. The `is` implementation is simply a single-element list of `in`...